### ISSUE TITLE
Help to compile on Windows

### ISSUE THREAD

**Issue by rodrigoncalves** [Svalorzen/asn1cpp#1]
Help me to make it work using Visual Studio. When I import the headers I receive some compilation errors. 

My code:
```
auto sigPolHash = asn1cpp::makeSeq(SigPolicyHash);
asn1cpp::setField(sigPolHash->hashAlgorithm, "1.2.840.113549.1.9.16.2.15");
asn1cpp::setField(sigPolHash->hashValue, "\x89\xA5\xEA\x68\xCC\x0F\x7E\x4E\x89\xA6\x23\xBF\xA6\xC8\x40\x75\xCB\xAF\x44\x35\x35\x5B\x82\x84\x56\xFF\xD1\x44\x8E\x5B\xA3\x52");
auto value = asn1cpp::getField(sigPolHash->hashAlgorithm, std::string);
``` 

Compilation error:
```
1>c:\project\include\asn1cpp\view.hpp(285): error C2976: 'asn1cpp::Impl::Setter': too few template arguments
1>c:\project\include\asn1cpp\view.hpp(14): note: see declaration of 'asn1cpp::Impl::Setter'
1>c:\project\include\asn1cpp\view.hpp(289): note: see reference to class template instantiation 'asn1cpp::View<T>' being compiled
1>c:\project\include\asn1cpp\setter.hpp(35): error C2976: 'asn1cpp::Impl::Setter': too few template arguments
1>c:\project\include\asn1cpp\setter.hpp(29): note: see declaration of 'asn1cpp::Impl::Setter'
1>c:\project\include\asn1cpp\setter.hpp(43): error C2913: explicit specialization; 'asn1cpp::Impl::Setter' is not a specialization of a class template
1>c:\project\include\asn1cpp\setter.hpp(59): error C2913: explicit specialization; 'asn1cpp::Impl::Setter' is not a specialization of a class template
1>c:\project\include\asn1cpp\setter.hpp(72): error C2913: explicit specialization; 'asn1cpp::Impl::Setter' is not a specialization of a class template
```

I am using Microsoft Windows 2017 with toolset v141.

**Comment by Svalorzen** [comment]
Can you post also the ASN1 definition of `SigPolicyHash`?

**Comment by Svalorzen** [comment]
Also, do the tests work for you?

**Comment by rodrigoncalves** [comment]
Well... I just change the line of code in View.hpp(284) from
```
template <typename Check>
friend struct Impl::Setter;
```
to
```
template <typename T, typename Check>
friend struct Impl::Setter;
```
and it works fine now.

I did not run the tests yet because I didn't configure Boost. But I can use this headers in my project now. Thank you for your attention! 😄

**Comment by Svalorzen** [comment]
That change will work although the resulting code is not quite the same. Hopefully it won't affect your code.

I'm looking into a way to fix it in the proper way in the meantime, thanks for the report!

**Comment by rodrigoncalves** [comment]
Ok I'll be waiting for that fix. Thanks!

**Comment by Svalorzen** [comment]
It actually seems that your solution is the only one possible, as C++ does not allow friends of partial specialization. Learned a new thing today.

Thanks again for the bug report, I'll add this fix ASAP!

**Comment by Svalorzen** [comment]
Fixed in 9eb340dfdfa48764d0b7ad62ea8f950cf83865dd.
